### PR TITLE
docs: update README

### PR DESCRIPTION
**Pull request by ivokub**
Add reference to gnark-crypto in the readme. We should also add the audit reports to the repository and link to them.

### CODE REVIEW

**Review by gbotrel — APPROVED** [review]

**Comment by ivokub** [comment]
@gbotrel - rerequesting review. I updated README a bit more. Does it make sense or is it too long now?

File: README.md [hunk 1]
```diff
@@ -44,6 +76,15 @@ You can also get in touch directly: gnark@consensys.net
 
 [Release Notes](CHANGELOG.md)
 
+## Audits
+
```
Comment (gbotrel) [inline README.md:80]: we may also add the one for worldcoin: https://leastauthority.com/wp-content/uploads/2023/08/Worldcoin_Groth16_Verifier_in_EVM_Smart_Contract_Final_Audit_Report.pdf

File: README.md [hunk 2]
```diff
@@ -5,8 +5,11 @@
 [![PkgGoDev](https://pkg.go.dev/badge/mod/github.com/consensys/gnark)](https://pkg.go.dev/mod/github.com/consensys/gnark)
 [![Documentation Status](https://readthedocs.com/projects/pegasys-gnark/badge/)][`gnark` User Documentation] [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.5819104.svg)](https://doi.org/10.5281/zenodo.5819104)
 
-`gnark` is a fast zk-SNARK library that offers a high-level API to design circuits. The library is open source and developed under the Apache 2.0 license
+`gnark` is a fast zk-SNARK library that offers a high-level API to design circuits. The library is open source and developed under the Apache 2.0 license.
 
+`gnark` uses [`gnark-crypto`] for the finite-field arithmetic and out-circuit implementation of cryptographic algorithms.
+
+`gnark` powers [`Linea zk-rollup`](https://linea.build).
```
Comment (gbotrel) [inline README.md:12]: that's fine. maybe we should have a dedicated section or .md page with list of known users, but it's a bit hard to track :) .

File: README.md [hunk 3]
```diff
@@ -24,11 +27,40 @@ To get started with `gnark` and write your first circuit, follow [these instruct
...
+  - proof engines - we compile the circuits, run the setup, prove and verify using native implementation
+  - Solidity verifier - in addition to the previous, we verify the proofs in Solidity verifier. See [`gnark-solidity-checker`]
+* regression testing - we have implemented [tests for reported issues](internal/regression_tests) to avoid regressions
+* constraint count testing - we have implemented [circuit size tests](internal/stats) to avoid regressions
+* serialization testing - we check that [serialization round-trip is complete](io/roundtrip.go)
+* side-effect testing - we check that circuit [compilation is deterministic](test/assert.go)
+* fuzz testing:
+  - circuit input fuzzing - we provide random inputs to the circuit to cause solver error
+  - native input fuzzing - we provide random inputs to various native methods to cause errors. We have also stored initial fuzzing corpus for regression tests.
+  - circuit definition fuzzing - we cooperate with Consensys Diligence to fuzz the circuit definitions to find bugs in the `gnark` circuit compiler.
```
Comment (gbotrel) [inline README.md:53]: as soon as @wuestholz blog post is out, we should highlight this and the continious fuzzing pipeline

**Review by gbotrel — APPROVED** [review]
lgtm, approved with minor comments

File: README.md (hunk 1 above)
Comment (ivokub) [reply README.md:80]: Added

File: README.md (hunk 3 above)
Comment (ivokub) [reply README.md:53]: Indeed, lets expand on this when the blog post comes out.

File: README.md (hunk 2 above)
Comment (ivokub) [reply README.md]: Added dedicated file.

**Review by gbotrel — APPROVED** [review]

**Pull request merged**